### DISCUSSION TITLE
No implicit conversion from ImmutableArray<T> to ReadOnlySpan<T> ?

### DISCUSSION THREAD

**Discussion by tfenise** [dotnet/runtime#88019] [General]
Why is there no implicit conversion from `ImmutableArray<T>` to `ReadOnlySpan<T>` ?

I found [this issue](https://github.com/dotnet/runtime/issues/22928), but somehow [the full API proposal](https://github.com/dotnet/runtime/issues/22928#issuecomment-855154372) missed the implicit conversion mentioned in the head of the issue.

`T[]` already has an implicit conversion to `ReadOnlySpan<T>`, so this is an inconsistency between `ImmutableArray<T>` and `T[]`. I understand that there is a risk of breaking overload resolution by adding an implicit conversion, though.

**Comment by bjorg** [comment]
I'm wondering the same thing.
